components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -3353,6 +3354,13 @@ public static NativeArray jsFunction_getSubscriptionsByApplication(Context cx,
                     PrivilegedCarbonContext.startTenantFlow();
                     PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
                 }
+
+                //check whether application exist prior to get subscription
+                if (!APIUtil.isApplicationExist(username, applicationName, groupingId)) {
+                    handleException("Application " + applicationName + " does not exist for user " +
+                            "" + username);
```

Comment:
There is an additional empty string ("") before the username. Is that needed? This could be added by the IDE when breaking the string into 2 lines but if it is not needed, please remove that